### PR TITLE
Support ninja

### PR DESCRIPTION
**Pull request by risemeup1**
### PR types

others
### PR changes

others
### Describe

在setup.py中增加ninja加速编译，如需指定不同的CMAKE_GENERATOR，export GENERATOR=xxx即可

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -557,12 +557,21 @@ def options_process(args, build_options):
             args.append("-D{}={}".format(key, value))
 
 
+def get_cmake_generators():
+    if os.getenv("CMAKE_GENERATOR"):
+        cmake_generators = os.getenv("CMAKE_GENERATOR")
```
Comment (zhiqiu) [inline setup.py]: cmake_generators -> cmake_generator

File: setup.py [hunk 2]
```diff
@@ -557,12 +557,21 @@ def options_process(args, build_options):
...
+    else:
+        cmake_generators = " Unix Makefiles"
```
Comment (zhiqiu) [inline setup.py]:
```suggestion
        cmake_generators = "Unix Makefiles"
```

File: setup.py (hunk 1 above)
Comment (risemeup1) [reply setup.py]: ok.thks

File: setup.py (hunk 2 above)
Comment (risemeup1) [reply setup.py]: ok,thks

File: setup.py [hunk 3]
```diff
@@ -596,83 +622,80 @@ def build_steps():
...
-                else:
-                    key = other_options[option_key]
-                if key not in paddle_build_options:
-                    paddle_build_options[key] = option_value
-        args = []
-        options_process(args, paddle_build_options)
-        print("args:", args)
-        cmake_run(args, build_path)
+
+    CMAKE_GENERATOR = get_cmake_generators()
```
Comment (zhangbo9674) [inline setup.py]: get_cmake_generators -> get_cmake_generator

Comment (risemeup1) [reply setup.py]: ok，thks

**Review by zhangbo9674 — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**